src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const str = sourceString;
+  const result = {};
+  const items = str.split(';').filter((item) => item.trim());
+
+  items.forEach((item) => {
+    const parts = item.split(':');
+    const key = parts[0].trim();
+    const value = parts.slice(1).join(':').trim();
```

Comment:
The use of `parts.slice(1).join(':')` ensures that any additional colons in the value part are preserved. This is important for maintaining the integrity of the value if it contains colons.